util/network/src/lib.rs

```diff
@@ -145,14 +148,61 @@ impl NonReservedPeerMode {
 	}
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "ipc", binary)]
+pub struct IpFilter {
+    pub predefined: AllowIP,
+    pub custom_allow: Vec<IpNetwork>,
+    pub custom_block: Vec<IpNetwork>,
+} 
+
+impl IpFilter {
+    pub fn new() -> IpFilter {
+        IpFilter {
+            predefined: AllowIP::All,
+            custom_allow: vec![],
+            custom_block: vec![],
+        }
+    }
+
+    /// Attempt to parse the peer mode from a string.
+    pub fn parse(s: &str) -> Result<IpFilter, IpNetworkError> {
+        let mut filter = IpFilter::new();
+        for f in s.split_whitespace() {
+            match f {
+                "all" => filter.predefined = AllowIP::All,
+                "private" => filter.predefined = AllowIP::Private,
+                "public" => filter.predefined = AllowIP::Public,
+                "none" => filter.predefined = AllowIP::Non,
+                custom => {
+                    if custom.starts_with("-") {
+                        match IpNetwork::from_str(&custom.to_owned().split_off(1)) {
+                            Ok(ipnet) => filter.custom_block.push(ipnet),
+                            Err(e) => { return Err(e); }
+                        }
+                    } else {
+                        match IpNetwork::from_str(custom) {
+                            Ok(ipnet) => filter.custom_allow.push(ipnet),
+                            Err(e) => { return Err(e); }
+                        }
+                    }
+                }
+            }
+        }
+        Ok(filter)
+    }
+}
+
 /// IP fiter
-#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum AllowIP {
 	/// Connect to any address
 	All,
 	/// Connect to private network only
 	Private,
 	/// Connect to public network only
 	Public,
+
+    Non,
```

Comment:
No, I didn't want it to be confused with `Option::None` but reading it now I can see that it might be more confusing :-p